src/Groups/group_constructors.jl

```diff
@@ -44,7 +44,7 @@ function symmetric_group(::Type{T}, n::Int) where T <: GAPGroup
   if n < 1
     throw(ArgumentError("n must be a positive integer"))
   end
-  return T(GAP.Globals.SymmetricGroup(_gap_filter(T), n))
+  return T(GAP.Globals.SymmetricGroup(_gap_filter(T), n))::T
```

Comment:
This should not be necessary resp. propably fixes the wrong problem. Better to use:
```suggestion
  return T(GAP.Globals.SymmetricGroup(_gap_filter(T), n)::GapObj)
```
This way Julia can prove which method for `T()` gets invoked, and thus correctly deduce the return type; this is better because now it has additional information, namely the `T` method being invoked, and can thus potentially perform further optimizations.